task/run-script-oci-ta/0.1/run-script-oci-ta.yaml

```diff
@@ -0,0 +1,246 @@
+
+        # Setting new namespace to run buildah - 2^32-2
+        echo 'root:1:4294967294' | tee -a /etc/subuid >>/etc/subgid
+
+        UNSHARE_ARGS=()
+        if [ "${HERMETIC}" == "true" ]; then
+          echo "The script will be executed with network isolation"
+          UNSHARE_ARGS+=("--net")
+        fi
+
+        cp -r /var/workdir/source /var/workdir/script
+        mkdir /var/workdir/output
+        buildah from --name script-container "$SCRIPT_IMAGE"
+        unshare -Uf "${UNSHARE_ARGS[@]}" --keep-caps -r --map-users 1,1,65536 --map-groups 1,1,65536 -w /var/workdir buildah run --volume /var/workdir/script:/var/workdir/script --volume /var/workdir/output:/var/workdir/output --workingdir "$SCRIPT_WORKDIR" --user root:root -- script-container "$@"
+
+        IMAGE_REFERENCE=$(buildah inspect script-container | jq -r '.FromImage + "@" + .FromImageDigest')
+        echo "$IMAGE_REFERENCE" > "$(results.SCRIPT_BASE_IMAGE.path)"
+
+        if [ -f "/var/workdir/output/output" ]; then
+          cat /var/workdir/output/output > "$(results.SCRIPT_OUTPUT.path)"
+        fi
+        if [ -f "/var/workdir/output/array" ]; then
+          cat /var/workdir/output/array > "$(results.SCRIPT_OUTPUT_ARRAY.array)"
+        fi
+      securityContext:
+        capabilities:
+          add:
+            - SETFCAP
+    - name: symlink-check
+      image: quay.io/konflux-ci/git-clone@sha256:4e53ebd9242f05ca55bfc8d58b3363d8b9d9bc3ab439d9ab76cdbdf5b1fd42d9
+      volumeMounts:
+        - mountPath: /var/workdir
+          name: workdir
+      env:
+        - name: PARAM_ENABLE_SYMLINK_CHECK
+          value: $(params.enableSymlinkCheck)
+        - name: CHECKOUT_DIR
+          value: $(params.SCRIPT_ARTIFACT_DIRECTORY)
+      script: |
+        #!/usr/bin/env bash
+        set -euo pipefail
+
+        check_symlinks() {
+          FOUND_SYMLINK_POINTING_OUTSIDE_OF_REPO=false
+          while read -r symlink; do
+            target=$(readlink -m "$symlink")
+            if ! [[ "$target" =~ ^$CHECKOUT_DIR ]]; then
+              echo "The cloned repository contains symlink pointing outside of the cloned repository: $symlink"
+              FOUND_SYMLINK_POINTING_OUTSIDE_OF_REPO=true
+            fi
+          done < <(find "$CHECKOUT_DIR" -type l -print)
+          if [ "$FOUND_SYMLINK_POINTING_OUTSIDE_OF_REPO" = true ]; then
+            return 1
+          fi
+        }
+
+        if [ "${PARAM_ENABLE_SYMLINK_CHECK}" = "true" ]; then
+          echo "Running symlink check"
+          check_symlinks
+        fi
+    - name: fix-permissions
+      image: quay.io/konflux-ci/buildah-task:latest@sha256:ab0ba3b70f99faa74d2dd737422a965197af4922dec0109113bc535a94db0dfd
+      volumeMounts:
+        - mountPath: /var/workdir
+          name: workdir
+      env:
+        - name: CHECKOUT_DIR
+          value: $(params.SCRIPT_ARTIFACT_DIRECTORY)
+      script: |
+        #!/usr/bin/env bash
+        set -euo pipefail
+
+        chown -R root:root "$CHECKOUT_DIR"
```

Comment:
Why need change permission?

Reply:
To integrate properly with buildah. `git-clone` clones with `uid: 0` and buildah expects that. Instead of running the script as root, the task runs the script with the pod's random ui and changes the permissions after.

If we don't change the permissions to align with git-clone buildah can fail because it can not write on the workdir later while running the build from this TA

Reply:
I see. Thanks for the explanation.

Reply:
That's a good idea. done